File: 2014/2014Robot/Subsystems/Chassis.cpp
#include "Chassis.h"
#include "../Robotmap.h"
#include "../Commands/DriveWithJoystickCommand.h"

Chassis::Chassis() : Subsystem("Chassis") {
	printf("[Chassis] Starting Construction\n");
	
	this->leftFrontMotor = new Jaguar(DRIVE_LEFT_FRONT_MOTOR);
	this->leftBackMotor = new Jaguar(DRIVE_LEFT_BACK_MOTOR);
	this->rightFrontMotor = new Jaguar(DRIVE_RIGHT_FRONT_MOTOR);
	this->rightBackMotor = new Jaguar(DRIVE_RIGHT_BACK_MOTOR);
	
	this->drive = new RobotDrive(this->leftFrontMotor,this->leftBackMotor, this->rightFrontMotor, this->rightBackMotor);
	//Todo Find out which motors need to be inverted.
	this->drive->SetInvertedMotor(RobotDrive::kFrontLeftMotor, true);
	this->drive->SetInvertedMotor(RobotDrive::kRearLeftMotor, true);
	this->drive->SetInvertedMotor(RobotDrive::kFrontRightMotor, true);
	this->drive->SetInvertedMotor(RobotDrive::kRearRightMotor, true);
	
	this->drive->SetSafetyEnabled(false);
	
	printf("[Chassis] Finished Construction\n");
}
    
void Chassis::InitDefaultCommand() {
	// Set the default command for a subsystem here.
	//SetDefaultCommand(new MySpecialCommand());
	SetDefaultCommand(new DriveWithJoystickCommand());
}


// Put methods for controlling this subsystem
// here. Call these from Commands.

void Chassis::Stop() {
	this->drive->ArcadeDrive(0.0,0.0);
}

void Chassis::ArcadeDrive(Joystick *stick) {
	this->drive->ArcadeDrive(stick);
}

void Chassis::ArcadeDrive(float move, float rotate) {
	this->drive->ArcadeDrive(move,rotate);
}

void Chassis::TankDrive(float left, float right) {
	this->drive->TankDrive(left, right);
}

void Chassis::TurnLeft(float rate) {
	this->drive->ArcadeDrive(0, rate);
}
void Chassis::TurnRight(float rate) {
	this->drive->ArcadeDrive(0, -rate);
}
RobotDrive *Chassis::GetRobotDrive() {
	return this->drive;
}
SpeedController *Chassis::GetLeftFrontMotor() {
	return this->leftFrontMotor;
}
SpeedController *Chassis::GetLeftBackMotor() {
	return this->leftBackMotor;
}
SpeedController *Chassis::GetRightFrontMotor() {
	return this->rightFrontMotor;
}
SpeedController *Chassis::GetRightBackMotor() {
	return this->rightBackMotor;
}

float Chassis::InchesToTicks(int inches) {
	//float grearRatio = 1.0;
	float wheelSize = 4.0;
	float ticksPerInch = 120.0 / (3.14 * wheelSize);
	return ticksPerInch * inches;
}

File: 2016/Prototypes/CollisionTest/src/com/kennedyrobotics/robot2016/subsystems/Chassis.java
package com.kennedyrobotics.robot2016.subsystems;

import edu.wpi.first.wpilibj.Jaguar;
import edu.wpi.first.wpilibj.Joystick;
import edu.wpi.first.wpilibj.RobotDrive;
import edu.wpi.first.wpilibj.SpeedController;
import edu.wpi.first.wpilibj.command.Subsystem;

import static com.kennedyrobotics.robot2016.RobotMap.*;

import com.kennedyrobotics.robot2016.commands.DriveWithJoystick;

/**
 *
 */
public class Chassis extends Subsystem {
    
	public static final double MAX_SPEED = 100;//Inches per second 
	
    // Put methods for controlling this subsystem
    // here. Call these from Commands.

	private final SpeedController frontLeft;
	private final SpeedController frontRight;
	
	private final RobotDrive drive;
	
    public void initDefaultCommand() {
        // Set the default command for a subsystem here.
        //setDefaultCommand(new MySpecialCommand());
    	setDefaultCommand(new DriveWithJoystick());
    }
    
    public Chassis() {
    	frontLeft = new Jaguar(CHASSIS_JAGUAR_FRONT_LEFT);
    	frontRight = new Jaguar(CHASSIS_JAGUAR_FRONT_RIGHT);
    	
    	drive = new RobotDrive(frontLeft, frontRight);

    	drive.setSafetyEnabled(false);
    }
    
    public void arcadeDrive(Joystick stick) {
    	double x = stick.getX();
    	double y = stick.getY();
    	
    	x = -x;
    	
    	drive.arcadeDrive(y,x);
    }
    
    public void arcadeDrive(double move, double rotate) {
    	drive.arcadeDrive(move, rotate);
    }
    
    public void tankDrive(double left, double right) {
    	drive.tankDrive(left, right);
    }
    
    public void setLeft(double speed) {
    	frontLeft.set(speed);
    }
    
    public void setRight(double speed) {
    	frontRight.set(speed);
    }
    
    public void stop() {
    	drive.tankDrive(0.0, 0.0);
    }    
}

